mypy_django_plugin/transformers/managers.py

```diff
@@ -232,17 +248,21 @@ def create_new_manager_class_from_from_queryset_method(ctx: DynamicClassDefConte
                 sym_type=AnyType(TypeOfAny.special_form),
             )
 
-    # we need to copy all methods in MRO before django.db.models.query.QuerySet
-    # Gather names of all BaseManager methods
+    # For methods that are common between BaseManager and QuerySet we need to
+    # update the return type of the methods on the manager to return the
+    # correct QuerySet type. This is done by adding the methods as attributes
+    # with type Any to the manager class, similar to how custom queryset
+    # methods are handled above. The actual type of these methods are resolved
+    # in resolve_manager_method.
+
+    # Find all methods that are defined on BaseManager
     manager_method_names = []
     for manager_mro_info in new_manager_info.mro:
         if manager_mro_info.fullname == fullnames.BASE_MANAGER_CLASS_FULLNAME:
-            for name, sym in manager_mro_info.names.items():
+            for name in manager_mro_info.names:
                 manager_method_names.append(name)
```

Comment:
Perhaps a micro-optimisation and out of scope. But `manager_method_names` could be a set, right?

```python
manager_method_names = set()
for manager_mro_info in new_manager_info.mro:
    if manager_mro_info.fullname == fullnames.BASE_MANAGER_CLASS_FULLNAME:
        for name in manager_mro_info.names:
             manager_method_names.add(name)
}
```

One could also make a comprehension out of it. Though I'm not sure if that's more, or less, readable..

```python
manager_method_names = {
    *manager_mro_info.names
    for manager_mro_info in new_manager_info.mro
    if manager_mro_info.fullname == fullnames.BASE_MANAGER_CLASS_FULLNAME
}
```

Reply:
Haha, I just did basically the same, but decided that it was less readable, so I changed it back.

I'm wondering if it would be better to have a simple predefined list of methods here though. It's fairly easy to define that list and it might be a better approach than trying to detect which methods return querysets. What do you think of that? Then we can use that same list to update the return types in `resolve_manager_method` and avoid potentially affecting custom queryset methods

Reply:
> What do you think of that?

I'd agree with that (I've had similar thoughts too). It's more difficult to understand what that `"_QS"` is there for..

Reply:
I mean, as long as there's a/some test cases testing that whole list. That should suffice for coverage and future changes.